### ISSUE TITLE
Sway workspace styles don't get removed consistently

### ISSUE THREAD

**Issue by iyzana** [Alexays/Waybar#2227]
The sway workspace styles (`.solo`, `.<appid>`, `.empty`, etc) are sometimes not removed anymore.

## (Preliminary) Steps to reproduce

I did not test this with a mostly empty config yet.

-  Add a style for `.firefox`, e.g.

```
window#waybar.firefox {
  background-color: #0088ff;
}
```

- Go to a new workspace
- Open two (tiled) windows (any application)
- Go to a new workspace
- Open firefox (the style should now be applied)
- Go back to the workspace with the two windows

## Expected Behavior

The style applied for `.firefox` is removed again

## Actual Behavior

The style applied for `.firefox` is still present

## Further info

This is a regression in `0.9.18`. `0.9.17` did not have this bug.

**Comment by RobertMueller2** [comment]
It's the window module and I can reproduce it for the appid styles, but not the others.

At first glance, I think something went wrong during one of the many rebase operations when I was working on it, but I could be mistaken. E.g. I don't think old_app_id_ should be set twice, but there it is in lines 183 and 204.

I'll see if I can fix it, but can't really promise a timeline.

**Comment by RobertMueller2** [comment]
I think I fixed the issue described in the repro. There was some leftover code I had previously replaced but at some point rebased it back in by accident.

If you do have issues with the other styles, that's likely a different code part and needs a different repro. I could not reproduce that.

**Comment by iyzana** [comment]
I wasn't very sure about the `.solo` and `.empty` styles not working. Could be that the `GTK_DEBUG=interactive` just didn't update properly.

I assume your fix is about what I reported. I tried to test your fix, but I can't get waybar to compile: `ninja -C build` is reporting some problems in `Catch2` (`catch_string_manip.hpp:47:14: error: ‘uint64_t’ in namespace ‘std’ does not name a type; did you mean ‘wint_t’?`). If its okay I'll just wait for the next release.
